File: Medium/1658. Minimum Operations to Reduce X to Zero/1658. Minimum Operations to Reduce X to Zero.py
class Solution:
    def minOperations(self, nums: List[int], x: int) -> int:
        total = sum(nums)
        target = total - x
        left = 0
        n = len(nums)
        maxLength = -1
        runningSum = 0

        for right in range(n):
            runningSum += nums[right]

            while runningSum > target and left <= right:
                runningSum -= nums[left]
                left += 1

            if runningSum == target:
                maxLength = max(maxLength, right - left + 1)

        return n - maxLength if maxLength != -1 else -1

File: Medium/1658. Minimum Operations to Reduce X to Zero/1658. Minimum Operations to Reduce X to Zero.rs
impl Solution {
    pub fn min_operations(nums: Vec<i32>, x: i32) -> i32 {
        let total: i32 = nums.iter().sum();
        let target = total - x;
        let mut left = 0;
        let mut running_sum = 0;
        let mut max_length = -1;

        for (right, &num) in nums.iter().enumerate() {
            running_sum += num;

            while running_sum > target && left <= right {
                running_sum -= nums[left];
                left += 1;
            }

            if running_sum == target {
                max_length = max_length.max((right - left + 1) as i32);
            }
        }

        if max_length == -1 {
            -1
        } else {
            nums.len() as i32 - max_length
        }
    }
}
